base/filesystem.jl

```diff
@@ -104,7 +104,7 @@ function sendfile(dst::File, src::File, src_offset::Int64, bytes::Int)
     nothing
 end
 
-function write(f::File, buf::Ptr{UInt8}, len::Integer, offset::Integer=-1)
+function unsafe_write(f::File, buf::Ptr{UInt8}, len::UInt, offset::Int64=Int64(-1))
```

Comment:
does this need to be tightened from Integer? this might break packages that are hitting code paths not tested in base